packages/calling/src/Voicemail/types.ts

```diff
@@ -70,22 +70,193 @@ export type VoicemailResponseEvent = {
+   * @param sort - Sort voicemail list (eg. ASC | DESC).
+   * @param refresh - Set to `true` to force a refresh of voicemail data from backend (optional).
+   *
+   * @example
+   * ```typescript
+   * const voicemailResponse = await voicemailInstance.getVoicemailList(0, 10, SORT.ASC);
+   * ```
+   * The `voicemailResponse` object will have `voicemailList` object as properties in data attribute.
+   *
+   * ```json
+   * {
+   *    statusCode: 200,
+   *    data: {
+   *        voicemailList: [messages]
+   *    }
+   * }
+   * ```
+   */
+  getVoicemailList(
     offset: number,
     offsetLimit: number,
     sort: SORT,
     refresh?: boolean
-  ) => Promise<VoicemailResponseEvent>;
-  getVoicemailContent: (messageId: string) => Promise<VoicemailResponseEvent>;
-  getVoicemailSummary: () => Promise<VoicemailResponseEvent | null>;
-  voicemailMarkAsRead: (messageId: string) => Promise<VoicemailResponseEvent>;
-  voicemailMarkAsUnread: (messageId: string) => Promise<VoicemailResponseEvent>;
-  deleteVoicemail: (messageId: string) => Promise<VoicemailResponseEvent>;
-  getVMTranscript: (messageId: string) => Promise<VoicemailResponseEvent | null>;
-  resolveContact: (callingPartyInfo: CallingPartyInfo) => Promise<DisplayInformation | null>;
+  ): Promise<VoicemailResponseEvent>;
+
+  /**
+   * Retrieves the content of a voicemail message based on its messageId.
+   * Received data can be accessed through `data.voicemailContent`
+   *
+   * @param messageId - The identifier of the voicemail message.
+   * @example
+   * ```typescript
+   * const messageId = 'Y2lzY29zcGFyazovL3VzL01FU1NBR0UvNTc3OTQ2NjItNDA5OS00NDQ3LWI';
+   * const voicemailResponse = await voicemailInstance.getVoicemailContent(messageId);
+   * ```
+   * The `voicemailResponse` object will have `voicemailContent` object as properties in data attribute.
+   *
+   * ```json
+   * {
+   *    statusCode: 200,
+   *    data: {
+   *        voicemailContent: {
+   *          type: 'message',
+   *          content: 'something'
+   *        }
+   *    }
+   * }
+   * ```
+   */
+  getVoicemailContent(messageId: string): Promise<VoicemailResponseEvent>;
+
+  /**
+   * Retrieves a quantitative summary of voicemails for a user.
+   * Received data can be accessed through `data.voicemailSummary`
+   *
+   * @example
+   * ```
+   * const voicemailResponse = await voicemailInstance.getVoicemailSummary();
+   * ```
+   * The `voicemailResponse` object will have `voicemailSummary` object as properties in data attribute.
+   *
+   * ```json
+   * {
+   *    statusCode: 200,
+   *    data: {
+   *        voicemailSummary: {
+   *          newMessages: 1,
+   *          oldMessage: 7,
+   *          newUrgentMessages: 0,
+   *          oldUrgentMessages: 0
+   *        }
+   *    }
+   * }
+   * ```
+   */
+  getVoicemailSummary(): Promise<VoicemailResponseEvent | null>;
+
+  /**
+   * Marks a voicemail message as read based on its message identifier.
+   * No data is returned in the response
```

Comment:
For APIs where no data is returned, we will still send the VoicemailResponseEvent and it will have statuscode and message set to success or failure based on what is returned from backend. So we should add that information